modules/govuk_jenkins/templates/jobs/deploy_app.yaml.erb

```diff
@@ -42,6 +42,27 @@
 
             git clone ${APP_DEPLOYMENT_GIT_URL} --branch master --single-branch --depth 1 ./
             ./jenkins.sh
+        - conditional-step:
+            condition-kind: and
+            condition-operands:
+              - condition-kind: or
+                condition-operands:
+                  <% @deploy_downstream_applications.keys.each do |app| %>
+                  - condition-kind: strings-match
+                    condition-string1: "$TARGET_APPLICATION"
+                    condition-string2: "<%= app %>"
+                  <% end %>
+                  - condition-kind: never
+              - condition-kind: regex-match
+                regex: "release_.*"
+                label: "$TAG"
+              - condition-kind: strings-match
+                condition-string1: "$DEPLOY_TASK"
+                condition-string2: "deploy"
```

Comment:
For readability, could you move this condition to be first, then the regex one, then finally the more complicated 'matching an application' condition? The `condition-kind: and` shortly followed by the `condition-kind: or` messed with my head and I think it would be easier to parse if they were simply reordered 😁 

Would there be a tiny efficiency benefit too, since the regex/string conditions are quicker to evaluate?

Reply:
Sure, I've reordered to have the simpler conditions first. I'm not sure the runtime of them is significant, though!